graylog2-web-interface/src/components/event-definitions/event-definition-form/field-value-providers/LookupTableFieldValueProviderForm.jsx

```diff
@@ -0,0 +1,130 @@
+    config: PropTypes.object.isRequired,
+    eventFields: PropTypes.object.isRequired,
+    lookupTables: PropTypes.array.isRequired,
+    onChange: PropTypes.func.isRequired,
+  };
+
+  static type = 'lookup-v1';
+
+  formatMessageFields = lodash.memoize(
+    (fieldTypes) => {
+      return fieldTypes
+        .sort((ftA, ftB) => naturalSort(ftA.name, ftB.name))
+        .map((fieldType) => {
+          return {
+            label: `${fieldType.name} – ${fieldType.value.type.type}`,
+            value: fieldType.name,
+          };
+        });
+    },
+    fieldTypes => fieldTypes.map(ft => ft.name).join('-'),
+  );
+
+  propagateChanges = (key, value) => {
+    const { config, onChange } = this.props;
+    const nextProviders = lodash.cloneDeep(config.providers);
+    const templateProvider = nextProviders.find(provider => provider.type === LookupTableFieldValueProviderForm.type);
+    templateProvider[key] = value;
+    onChange(Object.assign({}, config, { providers: nextProviders }));
+  };
+
+  handleChange = (event) => {
+    const { name } = event.target;
+    const value = FormsUtils.getValueFromInput(event.target);
+    this.propagateChanges(name, value);
+  };
+
+  handleSelectChange = (key) => {
+    return (nextLookupTable) => {
+      this.propagateChanges(key, nextLookupTable);
+    };
+  };
+
+  formatFields = (context) => {
+    const { allFieldTypes, eventFields } = this.props;
+
+    if (context === 'source') {
+      return this.formatMessageFields(allFieldTypes);
+    }
+
+    return Object.keys(eventFields)
+      .sort(naturalSort)
+      .map(fieldName => ({ label: fieldName, value: fieldName }));
+  };
+
+  formatLookupTables = (lookupTables) => {
+    return lookupTables.map(table => ({ label: table.title, value: table.name }));
+  };
+
+  render() {
+    const { config, lookupTables } = this.props;
+    const provider = config.providers.find(p => p.type === LookupTableFieldValueProviderForm.type);
+
+    return (
+      <Row className="row-sm">
+        <Col md={12}>
+          <FormGroup>
+            <ControlLabel>Take Lookup Table key from...</ControlLabel>
+            <Radio id="lookup-message-key-context"
+                   name="key_context"
+                   value="source"
+                   checked={provider.key_context === 'source'}
+                   onChange={this.handleChange}>
+              Source log message
+            </Radio>
+            <Radio id="lookup-event-key-context"
+                   name="key_context"
+                   value="event"
+                   checked={provider.key_context === 'event'}
+                   onChange={this.handleChange}>
+              Generated Event
+            </Radio>
+          </FormGroup>
+
+          <FormGroup controlId="lookup-provider-table">
+            <ControlLabel>Lookup Table Key Field</ControlLabel>
+            <Select name="lookup-provider-key"
+                    placeholder="Select Lookup Table"
```

Comment:
This should probably be "Select Field" or something similar.

Reply:
✅